### ISSUE TITLE
Positioning

### ISSUE THREAD

**Issue by seanohue** [shawncplus/ranviermud#168]
Implement positioning (necessary for some commands such as "sleep", could also be interesting to use a "knockdown" skill to change the positioning of other characters).

Basically, some commands will only be available in certain positions: laying, sitting, or standing, for example.

`sleep`, for instance, would give a huge boost to health/energy/mana regen but would change the player character's position, which would affect which commands are available and even affect the way they receive certain inputs (such as not being able to "hear" someone in the same room `say`ing something while they are asleep).

This could be implemented via having an attribute on the PC to describe their position (on all characters, in fact), commands to change their position (`stand` to recover from being knocked down before they can fight back, for example), and then a property on command objects to list any blacklisted positions (or whitelisted ones, either way is fine). For instance:

```
// look command config
return {
    usage: "look [thing]",
    blacklistedPositions: ["sleeping"],
    command: state => (args, player) => { } // etc. etc.
    }
```

This is just a suggestion as far as implementation goes, there is probably a better way 🌵 .

**Comment by shawncplus** [comment]
That's basically how I imagine it working but positions would just be an ascending numeric enum. `minimumPosition: Positions.RESTING`. `SLEEPING` = 0, RESTING = 1, STANDING = 2` so minimumPosition `RESTING` would just automatically include anything above resting

**Comment by marcelomoreli** [comment]
I would do some positions as well like DEAD, STUNNED, SITTING, FIGHTING

**Comment by shawncplus** [comment]
Yeah the plan is to have it like player roles where there are some base positions in the core but the bundles can extend/override

**Comment by marcelomoreli** [comment]
Perfect! Another observation, In my opinion this is useful for the NPCs as well, not only for players, so, my guess is that it would be better at Character level

**Comment by shawncplus** [comment]
Absolutely yeah. I could imagine stuff like getting a sneak attack bonus against sleeping NPCs or something
